### ISSUE TITLE
how to make sure code runs correctly?

### ISSUE THREAD

**Issue by Sandy4321** [shuhantao/KDD2019-Challenge#2]
I got this result , is it correct ?

----Start [train]:
--------remaining time 179
------------------ Training Params ------------
--------{'objective': 'binary', 'metric': 'auc', 'verbosity': -1, 'num_threads': 4}
--------Start [hyperopt_lgb]:

  0%|          | 0/60 [00:00<?, ?it/s, best loss: ?]
  2%|▏         | 1/60 [00:04<04:21,  4.44s/it, best loss: -0.7150881672288598]
  3%|▎         | 2/60 [00:06<03:36,  3.73s/it, best loss: -0.7292634724296472]
  5%|▌         | 3/60 [00:08<03:03,  3.23s/it, best loss: -0.7319374215003834]
  7%|▋         | 4/60 [00:11<02:55,  3.13s/it, best loss: -0.7319374215003834]
  8%|▊         | 5/60 [00:21<04:44,  5.18s/it, best loss: -0.735583160956043] 
 10%|█         | 6/60 [00:23<03:54,  4.35s/it, best loss: -0.735583160956043]
 12%|█▏        | 7/60 [00:25<03:03,  3.47s/it, best loss: -0.735583160956043]
 13%|█▎        | 8/60 [00:27<02:33,  2.95s/it, best loss: -0.735583160956043]
 15%|█▌        | 9/60 [00:31<02:55,  3.43s/it, best loss: -0.735583160956043]
 17%|█▋        | 10/60 [00:37<03:25,  4.11s/it, best loss: -0.7371157810336343]
 18%|█▊        | 11/60 [00:39<02:51,  3.50s/it, best loss: -0.7371157810336343]
 20%|██        | 12/60 [00:40<02:12,  2.75s/it, best loss: -0.7371157810336343]
 22%|██▏       | 13/60 [00:42<01:55,  2.46s/it, best loss: -0.7371157810336343]
 23%|██▎       | 14/60 [00:49<03:07,  4.07s/it, best loss: -0.7428314561374898]
------------auc = 0.7428 {'bagging_fraction': 0.8, 'bagging_freq': 30, 'feature_fraction': 0.8, 'learning_rate': 0.023187904810781, 'max_bin': 255, 'min_child_weight': 7.3023396420695255, 'num_leaves': 89, 'reg_alpha': 0.35147523167570927, 'reg_lambda': 1.4094707100947967}
{'bagging_fraction': 0.8, 'bagging_freq': 30, 'feature_fraction': 0.8, 'learning_rate': 0.023187904810781, 'max_bin': 255, 'min_child_weight': 7.3023396420695255, 'num_leaves': 89, 'reg_alpha': 0.35147523167570927, 'reg_lambda': 1.4094707100947967}
--------End   [hyperopt_lgb]. Time elapsed: 50.64 sec.
Important, only 97.73144626617432 for training
194439, 188888,0.11740262167794364
Important, only 304 for training
[LightGBM] [Info] Saving data to binary file train1584992901.1176152.bin
[LightGBM] [Info] Saving data to binary file valid1584992901.1176152.bin
Training until validation scores don't improve for 30 rounds
[100]	valid_0's auc: 0.744735
[200]	valid_0's auc: 0.746827
Early stopping, best iteration is:
[183]	valid_0's auc: 0.747949
--------[fitting] spend 76.97 sec
----End   [train]. Time elapsed: 76.97 sec.

----Start [predict]:
----End   [predict]. Time elapsed: 0.56 sec.
End   [predict]. Time elapsed: 199.49 sec.
INFO  [03-23 15:48:47] Prediction success, time spent so far 199.85722494125366 sec
INFO  [03-23 15:48:47] Done, exec_time=199.85722494125366
INFO  [03-23 15:48:47] Write results
